build/RepositoryBuild.targets

```diff
@@ -0,0 +1,69 @@
+<Project>
+  <Import Project="$(_BuildGraphFile)" />
+
+  <Target Name="BuildRepositories">
+    <ItemGroup>
+      <BatchedRepository Include="$(MSBuildProjectFullPath)">
+        <AdditionalProperties>
+          RepositoryToBuild=%(RepositoryToBuildInOrder.Identity);
+          BuildRepositoryRoot=$(_CloneRepositoryRoot)%(RepositoryToBuildInOrder.Identity)\
+        </AdditionalProperties>
+        <BuildGroup>%(RepositoryToBuildInOrder.Order)</BuildGroup>
+      </BatchedRepository>
+    </ItemGroup>
+
+    <PropertyGroup>
+      <BatchBuilds Condition="$(BuildInParallel) AND '$(CompileOnly)'=='true'">true</BatchBuilds>
+      <BatchBuilds Condition="'$(BatchBuilds)'==''">false</BatchBuilds>
+    </PropertyGroup>
+
+    <MSBuild
+      Projects="@(BatchedRepository)"
+      BuildInParallel="$(BatchBuilds)"
+      Targets="_BuildRepository"
+      Properties="BuildGroup=%(BatchedRepository.BuildGroup)" />
+  </Target>
+
+  <Target Name="_BuildRepository" DependsOnTargets="_PinVersions">
+    <PropertyGroup>
+      <BuildScriptToExecute Condition="'$(OS)'!='Windows_NT'">build.sh</BuildScriptToExecute>
```

Comment:
> Maybe that's an issue we can deal with when it arises.

That's a good concern. I was actually just thinking about that. I don't expect we will hit this soon. In the long run, I'm hoping we can do something like https://github.com/aspnet/KoreBuild/issues/196. In the short run, we could fall back to invoking build.cmd if we need to.